src/sirc/sirc.c

```diff
@@ -359,31 +359,46 @@ static void on_disconnect_ready(GObject *obj, GAsyncResult *result, gpointer use
 
 static void on_connect_finish(SircSession *sirc, GIOStream *stream){
     LOG_FR("Connected");
+    SircMessageContext *context = sirc_message_context_new(NULL);
 
     sirc->stream = stream;
     sirc_recv(sirc);
 
-    g_return_if_fail(sirc->events->connect);
-    sirc->events->connect(sirc, "CONNECT");
+    if (!sirc->events->connect) {
+        sirc_message_context_free(context);
```

Comment:
How about using [g_auto](http://www.manpagez.com/html/glib/glib-2.46.2/glib-Miscellaneous-Macros.php#g-auto) for freeing context?

Reply:
Oh nice, thanks! I'll use `g_autoptr` (which seems to be the right one here)

Reply:
There are other calls to sirc_message_context_free in the code, please check them : )

Reply:
including a couple of memory leaks, oops